vendor/go.etcd.io/etcd/client/v3/watch.go

```diff
@@ -580,6 +589,26 @@ func (w *watchGrpcStream) run() {
 
 			switch {
 			case pbresp.Created:
+				cancelReasonError := v3rpc.Error(errors.New(pbresp.CancelReason))
+				if shouldRetryWatch(cancelReasonError) {
```

Comment:
this looks really weird... is this handling a canceled flow in the created case? is that correct?

even if that's correct, it looks like this is triggering even in non-error cases (pbresp.CancelReason == "") - it's always creating a new error, a new rpc error, and calling `shouldRetryWatch` (I also commented in that method that stringification is happening more than it should)

Reply:
cc @ahrtr @serathius

Reply:
See my comment above https://github.com/kubernetes/kubernetes/pull/114403#discussion_r1046397696

Reply:
if pbresp.CancelReason is empty, shouldn't we avoid all of this added code?

Reply:
Resolved.